### PR TITLE
Improve the documentation for ActiveRecord::Inheritance#base_class

### PR DESCRIPTION
**Pull request by andrewn617**
### Motivation / Background

This Pull Request has been created to improve the documentation for `ActiveRecord::Inheritance#base_class`.

### Detail

While fixing https://github.com/rails/rails/pull/48991 I noticed that the documentation for `base_class` is somewhat ambiguous. Specifically this line:

```
Returns the class descending directly from ActiveRecord::Base, or an abstract class, if any, 
in the inheritance hierarchy.
```

This could be interpreted as "The class that descends from ActiveRecord::Base or that is an abstract class" (wrong) or "The class that descends from ActiveRecord::Base or that descends from an abstract class" (right).

I tried to improve the wording of this line. And I rewrote the examples as code with proper class names rather than the natural language examples that were there previously. I used the same domain for the examples as the docs for `abstract_class` to keep it consistent.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [ ] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Review by nvasilevski — APPROVED** [review]
👏

**Review by skipkayhil — APPROVED** [review]

File: activerecord/lib/active_record/inheritance.rb [hunk 1]
```diff
@@ -94,14 +94,22 @@ def finder_needs_type_condition? # :nodoc:
         :true == (@finder_needs_type_condition ||= descends_from_active_record? ? :false : :true)
       end
 
-      # Returns the class descending directly from ActiveRecord::Base, or
-      # an abstract class, if any, in the inheritance hierarchy.
+      # Returns the first class in the inheritance hierarchy that descends from either an
+      # abstract class or from ActiveRecord::Base
```
Comment (adrianna-chang-shopify) [inline activerecord/lib/active_record/inheritance.rb:98]:
```suggestion
      # abstract class or from +ActiveRecord::Base+.
```

**Review by adrianna-chang-shopify — COMMENTED** [review]
Small suggestion, I'll merge after.

File: activerecord/lib/active_record/inheritance.rb (hunk 1 above)
Comment (adrianna-chang-shopify) [reply activerecord/lib/active_record/inheritance.rb:98]: Oh actually, docs explicitly say to use `<tt></tt>` for `ActiveRecord::Base`, so let's use that instead: https://guides.rubyonrails.org/api_documentation_guidelines.html#fixed-width-font

**Pull request merged**